commit 5e7c1c918a4433c5dd1844d99f5648e17425fa16
Author: Matt Redfearn <matt.redfearn@mips.com>
Date:   Thu Mar 31 10:05:40 2016 +0100

    MIPS: Introduce plat_get_fdt a platform API to retrieve the FDT
    
    Early access to the kernel command line requires early access to the FDT
    for platforms which pass the command line within the device tree. There
    was no common way to retrieve the location of the FDT without incurring
    side effects, such as plat_mem_setup which, on Malta at least,
    initializes a bunch of other stuff.
    
    This patch adds plat_get_ftd() for IMG platforms.
    
    Signed-off-by: Matt Redfearn <matt.redfearn@imgtec.com>
    Cc: Paul Burton <paul.burton@imgtec.com>
    Cc: linux-mips@linux-mips.org
    Cc: kernel-hardening@lists.openwall.com
    Cc: linux-kernel@vger.kernel.org
    Patchwork: https://patchwork.linux-mips.org/patch/12988/
    Signed-off-by: Ralf Baechle <ralf@linux-mips.org>

diff --git a/arch/mips/mti-sead3/sead3-setup.c b/arch/mips/mti-sead3/sead3-setup.c
index e43f4801a245..9f2f9b2b23ce 100644
--- a/arch/mips/mti-sead3/sead3-setup.c
+++ b/arch/mips/mti-sead3/sead3-setup.c
@@ -83,6 +83,11 @@ static void __init parse_memsize_param(void)
 	}
 }
 
+void __init *plat_get_fdt(void)
+{
+	return (void *)__dtb_start;
+}
+
 void __init plat_mem_setup(void)
 {
 	/* allow command line/bootloader env to override memory size in DT */